solution_python.py

```diff
@@ -1,23 +1,80 @@
+from enum import Enum
+
+class Events(Enum):
+    ADD = 1
+    SUBTRACT = 2
+    MULTIPLY = 3
+    DIVIDE = 4
+    POW = 5
+
+class Event():
+    def __init__(self, value, change, event):
+        self.value = value
+        self.change = change
+        self.event = event
+
 class EventSourcer():
     # Do not change the signature of any functions
 
+    EVENT_FUNCTIONS = {
+        Events.ADD      : lambda x,y: x+y,
+        Events.SUBTRACT : lambda x,y: x-y,
+        Events.MULTIPLY : lambda x,y: x*y,
+        Events.DIVIDE   : lambda x,y: x//y,
+        Events.POW      : lambda x,y: int(x**y)
+    }
+
     def __init__(self):
         self.value = 0
+        self.current_event = -1
+        self.events = []
 
     def add(self, num: int):
```

Comment:
Looking back, I've got some code duplication here. If I thought of this earlier I would have done a helper method:
```python
def do_event(self, event_type, num):
    self.current_event += 1
    self.events.append(Event(self.value, num, event_type))
    func = EventSource.EVENT_FUNCTIONS[event_type]
    self.value = func(self.value, num)
```
Then each of the arithmetic methods would just be a single call to `do_event()` with the correct `Events` flag.